etl/dw/base_mongo.py

```
@@ -0,0 +1,116 @@
+            _list.extend([c])
+            if i % self.chunksize == 0:
+                yield _pd.DataFrame(_list)
+                _list = []
+
+        yield _pd.DataFrame(_list)
+
+    def run(self):
+        self.before_run()
+        self.extract()
+        self.extract_dimension()
+        for self.df_extract in self.itercursor():
```

Ruff (0.12.2)

Loop control variable `self` overrides iterable it iterates

(B020)